ReactWindows/ReactNative.Shared/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,229 @@
+        /// </summary>
+        /// <param name="element"></param>
+        /// <param name="value"></param>
+        public static void SetImportantForAccessibilityAttached(UIElement element, ImportantForAccessibility value)
+        {
+            element.SetValue(ImportantForAccessibilityAttachedProperty, value);
+        }
+
+        /// <summary>
+        /// ImportantForAccessibilityAttached property getter.
+        /// </summary>
+        /// <param name="element"></param>
+        /// <returns></returns>
+        public static ImportantForAccessibility GetImportantForAccessibilityAttached(UIElement element)
+        {
+            return (ImportantForAccessibility)element.GetValue(ImportantForAccessibilityAttachedProperty);
+        }
+
+        /// <summary>
+        /// Recursively sets AccessibilityView to accessibilityView for parentAutomationPeer children.
+        /// </summary>
+        /// <param name="parentAutomationPeer"></param>
+        /// <param name="accessibilityView"></param>
+        private static void SetChildrenAccessibilityView(AutomationPeer parentAutomationPeer, AccessibilityView accessibilityView)
+        {
+            if (parentAutomationPeer?.GetChildren() == null)
+            {
+                return;
+            }
+
+            foreach (AutomationPeer childAutomationPeer in parentAutomationPeer.GetChildren())
+            {
+                if (childAutomationPeer is FrameworkElementAutomationPeer childFEAutomationPeer)
+                {
+                    if (childFEAutomationPeer?.Owner != null)
+                    {
+                        AutomationProperties.SetAccessibilityView(childFEAutomationPeer.Owner, accessibilityView);
+                    }
+                }
+                else if (childAutomationPeer is ItemAutomationPeer childIAutomationPeer)
+                {
+                    if (childIAutomationPeer?.Item is UIElement childUIElement)
+                    {
+                        AutomationProperties.SetAccessibilityView(childUIElement, accessibilityView);
+                    }
+                }
+
+                SetChildrenAccessibilityView(childAutomationPeer, accessibilityView);
+            }
+        }
+
+        /// <summary>
+        /// Returns true if in uiElementAutomationPeer parents chain there is an UIElement with
+        /// ImportantForAccessibility value of Yes or NoHideDescendants.
+        /// </summary>
+        /// <param name="uiElementAutomationPeer"></param>
+        /// <returns></returns>
+        private static bool IsHiddenElement(AutomationPeer uiElementAutomationPeer)
+        {
+            if (uiElementAutomationPeer == null)
+            {
+                return false;
+            }
+
+            ImportantForAccessibility importantForAccessibilityAttached = ImportantForAccessibility.Auto;
+            var parentAutomationPeer = uiElementAutomationPeer.Navigate(AutomationNavigationDirection.Parent) as AutomationPeer;
```

Comment:
>Navigate [](start = 63, length = 8)

navigating to parent through automation peers prevents us from having the property set on non-UIElement elements like TextElement descendants (Run, Span and other inlines), Flyouts, Window. Maybe we want to preserve this capability and instead switch to elements themselves for Parent walk instead of through peers?

Reply:
I thought about this but my opinion is that the navigation should follow parent/child chain that is available via AutomationPeer because this is what narrator sees. I think that Parent property is not reliable to follow the way up since I've seen instances when it is set to null and that breaks the ability to navigate above certain point. We can discuss this

---
In reply to: [163117765](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r163117765) [](ancestors = 163117765)